packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts

```diff
@@ -430,7 +431,24 @@ export class ScheduleTrigger implements INodeType {
 				Second: moment.tz(timezone).format('ss'),
 				Timezone: moment.tz(timezone).format('z Z'),
 			};
-			this.emit([this.helpers.returnJsonArray([resultData])]);
+			const weeklyExecution = i ? (staticData[i] as IDataObject) : ({} as IDataObject);
+
+			// Checks if have the right week interval, handels new years as well
+			if (
+				isWeekly &&
+				weeklyExecution &&
+				i &&
```

Comment:
Same here, we need to check for `i !== undefined` otherwise when `i = 0` it won't work correctly